Gemfile

```diff
@@ -1,4 +1,12 @@
-source "http://rubygems.org"
+source 'http://rubygems.org'
+
+if ENV['BUILD_ROLE'] == 'ruby27'
+  ruby '2.7.2'
+elsif ENV['BUILD_ROLE'] == 'ruby23'
+  ruby '2.3.3'
+else
+  ruby '2.5.8'
+end
```

Comment:
Given this is a public gem, I'm not sure if we should add in explicit ruby versions.

Since the ruby version was missing here, can we still support our matrix build without specifying a ruby version in our gem file?

Reply:
This should be reflected by the s.required_ruby_version I added to our gemspec. The Gemfile is not packaged as a part of the gem, so I consider explicit versions here for development and testing purposes to be okay.

Reply:
I am totally open to removing the s.required_ruby_version from the gemspec if you truly want is to leave things wide open. My thoughts here we're to steer developers towards using a ruby that's actively in support.

Reply:
`required_ruby_version` specifies the base version this gem supports, which is great.

My issue is more with this requiring one of these three ruby versions to be installed/active when pulling down this gem for development.

With this removed, wouldn't the ruby version available on the executor just be used, based on the build role, making the 'version selection' (so to speak) self correcting?  In other words, on build-ruby27, ruby 2.7.X would be the ruby version used to build the gemfile, even without specifying it like so?

Reply:
Great point. I'll update accordingly!